pages/[slug].js

```diff
@@ -0,0 +1,155 @@
+import sportLocationsData from "../lib/data/sportLocationsData";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { IoStar, IoLocationSharp, IoChevronBackOutline } from "react-icons/io5";
+import CardLocationInfo from "../components/CardLocationInfo/CardLocationInfo";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const slug = router.query.slug;
+  const currentLocation = sportLocationsData.find(
+    (location) => location.slug === slug
+  );
+  const locationAddress = currentLocation?.address;
+
+  if (!currentLocation) {
+    return (
+      <>
+        <h2>Sorry, this page does not exist!</h2>
+      </>
+    );
+  }
+
+  return (
+    <>
+      <div>
```

Comment:
```suggestion
```
Either a wrapper div or a fragment (better a fragment), but in this case you don`t need either because your Main is the only parent element in the return statement :)